lenskit-core/src/main/java/org/lenskit/data/entities/AbstractEntity.java

```diff
@@ -57,6 +57,16 @@ public EntityType getType() {
     /**
      * {@inheritDoc}
      *
+     * This implementation delegates to {@link #hasAttribute(String)}
+     */
+    @Override
+    public boolean hasAttribute(TypedName<?> name) {
+        return hasAttribute(name.getName());
```

Comment:
Good question. I can go either way on it. It  is not hard to add a type check.

Right now, the API def maintains a relationship that the `get` methods throw `NoSuchAttributeException` if and only if `hasAttribute` returns false. If we make this check type and return `false` on a mismatch, then if `hasAttribute` returns `false` then `get` might throw `NoSuchAttributeException`, or it might throw `IllegalArgumentException`.